### PR TITLE
add explanation on what the bb_runner container situation is about

### PR DESCRIPTION
**Pull request by lucasmeijer**
Add more information about why there are two bb_runner container images, and when and how to use them.

### CODE REVIEW

**Review by EdSchouten — COMMENTED** [review]
LGTM, but can you please wrap sentences at <80 columns for consistency with the rest of the document?

**Review by EdSchouten — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -41,14 +41,26 @@ following:
...
+userland to be installed.
+
+It would be nice if you could just use any image of your choosing as the image
+that your build actions will run on. Like using
+[Ubuntu 16.04 image](https://console.cloud.google.com/marketplace/details/google/rbe-ubuntu16-04),
+to take advantage of the fact that bazel project provides [ready-to-use toolchain 
+definitions](https://github.com/bazelbuild/bazel-toolchains) for them.
+
+What makes that tricky is that that image will not have `bb_runner` installed.
+This  is where `bb_runner_installer` image comes in. It doesn't actually
```
Comment (moroten) [inline README.md:56]:
```suggestion
This is where `bb_runner_installer` image comes in. It doesn't actually
```

File: README.md [hunk 2]
```diff
@@ -41,14 +41,26 @@ following:
...
+install anything, but it provides the `bb_runner` executable through its
+filesystem. You have to configure your orchestration of choice to mount this
+filesystem from `bb_runner_installer` into the image of your choice that you
+want to run on. This way you can use a vanilla image and just run the bb_runner
+executable from buildbarn's provided container. There's a few tricks to check
+if the volume is already available, you can see an example of how to do this
+in the [docker-compose example](https://github.com/buildbarn/bb-deployments/blob/e404c1a519355353d0e2cdfd447126fe07095594/docker-compose/docker-compose.yml#L89)
```
Comment (moroten) [inline README.md:63]:
```suggestion
in the [docker-compose example](https://github.com/buildbarn/bb-deployments/blob/e404c1a519355353d0e2cdfd447126fe07095594/docker-compose/docker-compose.yml#L89).
```

File: README.md [hunk 3]
```diff
@@ -41,14 +41,26 @@ following:
...
+to take advantage of the fact that bazel project provides [ready-to-use toolchain 
+definitions](https://github.com/bazelbuild/bazel-toolchains) for them.
+
+What makes that tricky is that that image will not have `bb_runner` installed.
+This  is where `bb_runner_installer` image comes in. It doesn't actually
+install anything, but it provides the `bb_runner` executable through its
+filesystem. You have to configure your orchestration of choice to mount this
+filesystem from `bb_runner_installer` into the image of your choice that you
+want to run on. This way you can use a vanilla image and just run the bb_runner
+executable from buildbarn's provided container. There's a few tricks to check
```
Comment (moroten) [inline README.md:61]:
```suggestion
executable from Buildbarn's provided container. There's a few tricks to check
```

File: README.md [hunk 4]
```diff
@@ -41,14 +41,26 @@ following:
...
-For `bb_runner`, it provides one image without a userland, and one that
-installs the `bb_runner` into another container on startup. The former
-is sufficient for [BuildStream](https://buildstream.build), while the
-latter can, for example, be used in combination with Google RBE's
-[Ubuntu 16.04 image](https://console.cloud.google.com/marketplace/details/google/rbe-ubuntu16-04).
-The advantage of using the Ubuntu 16.04 image is that the Bazel project
-provides [ready-to-use toolchain definitions](https://github.com/bazelbuild/bazel-toolchains)
-for them.
+For `bb_runner`, it provides two images: `bb_runner_bare` and `bb_runner_installer`.
+`bb_runner_bare` has no userland/linux install, it just has the bb_runner
```
Comment (moroten) [inline README.md:45]:
```suggestion
`bb_runner_bare` has no userland/linux install, it just has the `bb_runner`
```

**Review by moroten — CHANGES_REQUESTED** [review]
Excellent. Just a few nit picks.

**Review by moroten — APPROVED** [review]

**Comment by stagnation** [comment]
Great thank you!

**Pull request merged**